Assets/Scripts/Modules/Turn/ITurnModule.cs

```diff
@@ -1,8 +1,17 @@
 ﻿/*
- * Authors: Xijie Guo, Isaiah Mann
+ * Authors: Xijie Guo
  * Description: An interface to handle the turn logic
  */
 
-public interface ITurnModule {
+public interface ITurnModule { 
 
+	void SwitchToMonTurn();
+
+	bool CheckMonsterAttack(IUnit monster, IUnit player);
```

Comment:
Checking if a monster can attack is in Combat module currently. Also handling if its magic or melee in the turn module is unimportant because magic monster cant melee and vice versa. At least that's my understanding so far.